aten/src/ATen/core/dispatch/KernelRegistration.h

```diff
@@ -80,58 +156,58 @@ class KernelRegistrar final {
  * @tparam OpSchemaDef The operator schema this is building a KernelRegistration for
  * @tparam FieldsPresentFlags Remembers which fields are already set in the builder
  */
-template<class OpSchemaDef, class StateTypeOrVoid, uint64_t FieldsPresentFlags>
+template<class CacheTypeOrVoid, uint64_t FieldsPresentFlags>
 class KernelRegistrationBuilder final {
 private:
-  using Schema = OpSchema<OpSchemaDef>;
-
   static constexpr uint64_t DISPATCH_KEY_PRESENT = 0x01 << 0;
   static constexpr uint64_t KERNEL_PRESENT = 0x01 << 1;
-  static constexpr uint64_t STATE_PRESENT = 0x01 << 2;
+  static constexpr uint64_t CACHE_PRESENT = 0x01 << 2;
 
-  static std::unique_ptr<c10::KernelState> defaultStateCreator() {
+  static std::unique_ptr<c10::KernelCache> defaultCacheCreator() {
     return nullptr;
   }
 
-  template<class State>
-  static std::unique_ptr<c10::KernelState> stateCreator() {
-    static_assert(std::is_default_constructible<State>::value, "State class must be default constructible");
-    return guts::make_unique<State>();
+  template<class Cache>
+  static std::unique_ptr<c10::KernelCache> cacheCreator() {
+    static_assert(std::is_default_constructible<Cache>::value, "Cache class must be default constructible");
+    return guts::make_unique<Cache>();
   }
 
-  c10::optional<typename Schema::dispatch::dispatch_key_type> dispatch_key_;
+  const OperatorHandle& (*op_)();
```

Comment:
Can we have a `using` declaration for the function type here please? :)

Reply:
For this, and the two other pointer fields below, can you please default initialize them to `= nullptr` explicitly? It will help me sleep better at night.

Reply:
They are already, the constructors are a few lines further below.